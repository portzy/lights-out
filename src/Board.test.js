import React from "react";
import { render } from "@testing-library/react";
// import '@testing-library/jest-dom/extend-expect'; 
import Board from "./Board";

test("renders Board without crashing", () => {
    render(<Board />);
  });

test("renders the starter board", () => {
  jest.spyOn(Math, "random").mockReturnValue(0.5); 
  const { container } = render(<Board nrows={3} ncols={3} chanceLightStartsOn={0.5} />);
  expect(container).toMatchSnapshot();
  Math.random.mockRestore(); 
});
