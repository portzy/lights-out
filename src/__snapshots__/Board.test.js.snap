// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the starter board 1`] = `
<div>
  <div>
    You Win!
  </div>
</div>
`;
